### PR TITLE
Adding multifunction support for nic adaptors while passthrough to guest

### PR DESCRIPTION
**Pull request by prudmirya**
Existing test case supports single function pass-through for nic and storage adaptor, now i am adding multifunction support for nic adaptors.
Signed-off-by: prudhvi <mprudhvi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by prudmirya** [comment]
job log added
[job.txt](https://github.com/autotest/tp-libvirt/files/1302157/job.txt)

File: libvirt/tests/src/libvirt_pci_passthrough.py [hunk 1]
```diff
@@ -128,33 +146,58 @@ def run(test, params, env):
...
+                    nic_name = str(utils_misc.get_interface_from_pci_id(val, session))
+                    try:
+                        session.cmd("ip addr flush dev %s" % nic_name)
+                        session.cmd("sleep 5")
+                        session.cmd("ip addr add %s/%s dev %s"
+                                    % (net_ip, netmask, nic_name))
+                        session.cmd("sleep 5")
+                        session.cmd("ip link set %s up" % nic_name)
+                        session.cmd("ifconfig %s" % nic_name)
+                        session.cmd("sleep 5")
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_pci_passthrough.py]: I dont think we need sleep time to configure using ip and ifconfig command

File: libvirt/tests/src/libvirt_pci_passthrough.py [hunk 2]
```diff
@@ -20,8 +21,8 @@ def run(test, params, env):
         1. Get params.
         2. Get the pci device for specific net_name.
         3. Attach pci device to guest.
-        4. Start guest and set the ip of guest.
-        5. Ping the server_ip of from guest
+        4. Start guest and set the ip to all the functions.
+        5. Ping to server_ip from each function
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_pci_passthrough.py]: please have clear doc string about functions, ie physical/virtual functions.

File: libvirt/tests/src/libvirt_pci_passthrough.py [hunk 3]
```diff
@@ -128,33 +145,59 @@ def run(test, params, env):
...
+                               "to ping server ip from guest. %s \n" % detail)
+            else:
+                output = session.cmd_output("lspci -nn | grep %s" % device_name)
+                nic_list = output.splitlines()
+                for val in range(len(nic_list)):
+                    bus_info.append(str(nic_list[val]).split(' ', 1)[0])
+                    nic_list[val] = str(nic_list[val]).split(' ', 1)[0][:-2]
+                # check all functions get same iommu group
+                if len(set(nic_list)) != 1:
+                    test.fail("Multifunction Device passthroughed but functions are in different iommu group")
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_pci_passthrough.py]:
```python
test.fail("Multifunction Device passthroughed"
          "but functions are in different iommu group")
```

**Comment by balamuruhans** [comment]
Thanks @prudmirya

**Pull request merged**